### ISSUE TITLE
Improve accessibility ( a11y )

### ISSUE THREAD

**Issue by bobbingwide** [bobbingwide/SB#46]
While working with my ex boss on the subject of accessibility ( a11y ) I ran the WebAIM WAVE browser extension and axe Dev Tools against several of my websites. seriouslybonkers.com, running the SB theme had 
- 3 WAVE errors
- 146 WAVE contrast errors
- 34 WAVE alerts
- 173 axe errors in total
- 144 axe contrast errors

### Requirements 

- Improve a11y to reduce WAVE errors and WAVE contrast errors to as close as possible to 0.
- Secondary requirement - explain how to achieve this
- Optionally - raise a11y concerns with 3rd parties

**Comment by bobbingwide** [comment]
## Colour contrast

### Link colours

The first problem was the link colour. 
- `#62add5` was too light for a white `#ffffff` background.

<img width="352" height="56" alt="Image" src="https://github.com/user-attachments/assets/f1dc0a8b-dc08-4090-8411-c59bf3400013" />

- I changed it to `#2c7ba5` to satisfy WCAG AAA.




- This reduced the contrast errors to 4.
- I then discovered that this darker link colour was still not dark enough when used with a background colour of `#f7f7f7` - a light grey

<img width="220" height="78" alt="Image" src="https://github.com/user-attachments/assets/24827f3e-4f39-408a-94a2-6f4ce7437d0a" />

- Changing the link colour to `#205a79` resolved this problem.


<img width="207" height="76" alt="Image" src="https://github.com/user-attachments/assets/0a8c6b63-a687-42b1-b28e-0bde9538ebfe" />

### Pagination links

- Next I found that the pagination link were not dark enough `#328db5`
- So I changed them to the same darker blue.

<img width="306" height="61" alt="Image" src="https://github.com/user-attachments/assets/dcf8f21b-0d83-40ff-a679-1197e9d613b4" />

- That still left me with 4 contrast errors

**Comment by bobbingwide** [comment]
### Social media links
Using WAVE I was unable to navigate to the 4 remaining contrast errors. But I could see that they were associated with the social links.

<img width="1583" height="844" alt="Image" src="https://github.com/user-attachments/assets/c77a969b-bb88-4955-850a-bbb3da8b0bde" />

- In my `footer` template part I'd embedded a `social-links` template part which uses the WordPress Social Icons block with Styles set to Default. 
- This contained 8 different social links for Twitter, Facebook, LinkedIn, Pinterest, Instagram, GitHub, WordPress and YouTube.

<img width="832" height="151" alt="Image" src="https://github.com/user-attachments/assets/a73735f6-4d4b-4ae1-bad9-9b49a5935b1f" />

- 4 of the icons produced contrast errors: Twitter, Instagram, WordPress and YouTube.
- I wasn't sure what I'd get with the Show text toggled on so tried that. 

<img width="1339" height="131" alt="Image" src="https://github.com/user-attachments/assets/9e3ee13b-493e-4daf-9764-b345408562ad" />

- the result is that the contrast errors are now visible... but I also get 8 Very Small Text alerts.
- The other thing is that I get ARIA hidden messages.

<img width="354" height="237" alt="Image" src="https://github.com/user-attachments/assets/2fedd5bd-efc6-4993-b429-18c91ccb177e" />

- Are ARIA counts good or bad? How can you tell?

**Comment by bobbingwide** [comment]
### Social media links continued

> 4 of the icons produced contrast errors: Twitter, Instagram, WordPress and YouTube.

These are examples of the _corporate color palette_ problem where the background colour matches the corporate brand colour.

- First thing to do is to replace Twitter with X
- Then check whether or not the other brand's colours have changed
-  https://brand.youtube/#color. Red is now `#ff0033`

What I actually did was to 
- change Twitter to X, 
- toggle off the Show text and
- Change the Size from Small to Normal

Then I tested with WAVE again. This time 0 Colour contrast errors from WAVE.

**Comment by bobbingwide** [comment]
But axe Dev Tools is having a moan with **Links must be distinguishable without relying on color**
suggesting two possible fixes...

- - - -
To solve this problem, you need to fix at least (1) of the following:
- The link has insufficient color contrast of 2.28:1 with the surrounding text. (Minimum contrast is 3:1, link text: `#205a79`, surrounding text: `#1a1a2a`)
- The link has no styling (such as underline) to distinguish it from the surrounding text

- - - -

**Comment by bobbingwide** [comment]
While investigating a comment made by a blind user regarding navigating blocks in the block editor we came across a problem which appears to be associated with blocks which have a specific Style set. When using the up/down arrow keys you can enter but can't leave a paragraph block which has a non Default style. Is this a known problem?
